test/common/http/access_log/access_log_formatter_test.cc

```diff
@@ -5,28 +5,53 @@
 
+using testing::NiceMock;
 using testing::Return;
 using testing::ReturnRef;
 
 namespace Http {
 namespace AccessLog {
 
 TEST(FailureReasonUtilsTest, toShortStringConversion) {
-  std::vector<std::pair<FailureReason, std::string>> expected = {
-      std::make_pair(FailureReason::None, "-"),
-      std::make_pair(FailureReason::FailedLocalHealthCheck, "LH"),
-      std::make_pair(FailureReason::NoHealthyUpstream, "UH"),
-      std::make_pair(FailureReason::UpstreamRequestTimeout, "UT"),
-      std::make_pair(FailureReason::LocalReset, "LR"),
-      std::make_pair(FailureReason::UpstreamRemoteReset, "UR"),
-      std::make_pair(FailureReason::UpstreamConnectionFailure, "UF"),
-      std::make_pair(FailureReason::UpstreamConnectionTermination, "UC"),
-      std::make_pair(FailureReason::UpstreamOverflow, "UO"),
-      std::make_pair(FailureReason::NoRouteFound, "NR"),
-      std::make_pair(FailureReason::FaultInjected, "FI")};
+  std::vector<std::pair<ResponseFlag, std::string>> expected = {
+      std::make_pair(ResponseFlag::None, "-"),
+      std::make_pair(ResponseFlag::FailedLocalHealthCheck, "LH"),
+      std::make_pair(ResponseFlag::NoHealthyUpstream, "UH"),
+      std::make_pair(ResponseFlag::UpstreamRequestTimeout, "UT"),
+      std::make_pair(ResponseFlag::LocalReset, "LR"),
+      std::make_pair(ResponseFlag::UpstreamRemoteReset, "UR"),
+      std::make_pair(ResponseFlag::UpstreamConnectionFailure, "UF"),
+      std::make_pair(ResponseFlag::UpstreamConnectionTermination, "UC"),
+      std::make_pair(ResponseFlag::UpstreamOverflow, "UO"),
+      std::make_pair(ResponseFlag::NoRouteFound, "NR"),
+      std::make_pair(ResponseFlag::FaultInjected, "FI"),
+      std::make_pair(ResponseFlag::DelayInjected, "DI")};
 
   for (const auto& testCase : expected) {
-    EXPECT_EQ(testCase.second, FilterReasonUtils::toShortString(testCase.first));
+    NiceMock<MockRequestInfo> request_info;
+    ON_CALL(request_info, isSetResponseFlag(testCase.first)).WillByDefault(Return(true));
+    EXPECT_EQ(testCase.second, FilterReasonUtils::toShortString(request_info));
+  }
+
+  // Test combinations.
+  {
+    NiceMock<MockRequestInfo> request_info;
+    ON_CALL(request_info, isSetResponseFlag(ResponseFlag::DelayInjected))
+        .WillByDefault(Return(true));
+    ON_CALL(request_info, isSetResponseFlag(ResponseFlag::UpstreamRequestTimeout))
+        .WillByDefault(Return(true));
+    EXPECT_EQ("UT,DI", FilterReasonUtils::toShortString(request_info));
+  }
+
+  {
+    NiceMock<MockRequestInfo> request_info;
+    ON_CALL(request_info, isSetResponseFlag(ResponseFlag::DelayInjected))
+        .WillByDefault(Return(true));
+    ON_CALL(request_info, isSetResponseFlag(ResponseFlag::UpstreamRequestTimeout))
+        .WillByDefault(Return(true));
+    ON_CALL(request_info, isSetResponseFlag(ResponseFlag::FaultInjected))
+        .WillByDefault(Return(true));
+    EXPECT_EQ("UT,FI,DI", FilterReasonUtils::toShortString(request_info));
```

Comment:
this was mostly to test basic functionality for multiple flags, i can place comment that it's not real use case but to test functionality